docs/historias_usuario.md

```diff
@@ -0,0 +1,19 @@
+## Historias de Usuario (User Stories)
+
+### [HU1]: Ana, Analista de Riesgos Financieros
+Soy Ana, analista de riesgos financieros en una empresa de banca. A diario, al evaluar las solicitudes de préstamos, me encuentro con la dificultad de que cada analista utiliza criterios diferentes para calcular el riesgo de los clientes. Esto no solo retrasa el proceso de decisión, sino que también provoca inconsistencias que pueden afectar la rentabilidad de la empresa. Además, el Banco de España exige que sigamos un criterio estándar y transparente para nuestras evaluaciones, lo que actualmente no se está cumpliendo de manera uniforme. Necesito un sistema que me permita estandarizar la manera de calcular el riesgo de cada operación, utilizando los datos del prestatario de forma automatizada para asegurar que todos los analistas utilicen los mismos criterios y procedimientos en sus evaluaciones.

```

Comment:
Has hecho una explicación del problema bastante completa. Pero la cuestión es ¿crees que una persona que lea esto puede empezar a trabajar en el problema? Sólo en la segunda frase ya encuentro diferentes palabras clave
- "Solicitud" de préstamo
- "Criterios" 
- "Riesgo"

Si quieres realmente que la persona que trabaje en esto haga una modelización de algo que te pueda resultar útil, tendrás que dividir el problema en piezas más pequeñas, que al fin y al cabo es la clave de este objetivo. Si una persona se cuentra con esos tres términos para modelar, y ni siquiera ha avanzado la tercera parte de la HU, va a ser imposible que avance ante la avalancha de términos a modelizar y definir.

Por otro lado, veo también una posibilidad que deberías considerar: la de extraer información de fuentes poco estructuradas. ¿El cliente te presenta directamente la solicitud en JSON? ¿Los criterios están también como expresiones de un lenguaje de programación? ¿Las "exigencias" del Banco de España también son directamente programas en el lenguaje de programación elegido?

Reply:
Como le comentaba anteriormente, añadí un [fichero](https://github.com/jacarmona364/Riskalc/blob/main/docs/material_financiero/essential_knowledge.md) con diferentes términos y nociones básicas sobre conceptos económicos, entre los que aparece por ejemplo "riesgo". No considero que las palabras "solicitud" o "criterio" requieran de una explicación más detallada, pues son de un nivel gramático poco sofisticado y al alcance de cualquiera, pero podría incluirlas sin problema. Mi intención no es abrumar con términos complejos sino eliminar cualquier ápice de duda del lector describiendo detalladamente el problema. ¿El camino a seguir debería ser rebajar la cantidad de información o la complejidad léxica del mismo? 

En cuanto a la fuente de información para el hito primero, es una serie de [datos](https://github.com/jacarmona364/Riskalc/blob/main/docs/material_financiero/historial_prestamos.csv) sobre operaciones pasadas que incluí en la documentación del proyecto. En el siguiente hito sí será necesario rellenar un formulario con los parámetros que se usan en ese mismo archivo.

Las exigencias del Banco de España son simplemente que se unifique el criterio de evaluación, sin entrar en la forma de hacerlo.

Reply:
El camino a seguir es dividir el problema. No se trata de rebajar el nivel léxico, se trata de explicar en qué consiste cada cosa para poder modelizarla y eventualmente programarla, y de dividir el problema en diferentes partes, de forma que se puedan ir creando productos de complejidad creciente. ¿De veras crees que cualquier pesona sólo con leer "solicitud de préstamo" sabe qué es lo que incluye la misma y que no necesita más explicación? ¿O "riesgo"? 

¿Crees de verdad que con definiciones como:

> Un préstamo es un acuerdo en el cual una entidad (por ejemplo, un banco o una persona) otorga una cantidad de dinero a otra parte, llamada prestatario, bajo la promesa de que será devuelto en un plazo establecido, generalmente con un interés adicional. El monto prestado suele denominarse capital y el contrato establece las condiciones, incluyendo el tipo de interés, los plazos de pago y cualquier penalización por incumplimiento.

se puede poner uno a trabajar? ¿Qué es un acuerdo? ¿Qué es una entidad? ¿Qué es una cantidad de dinero? ¿Qué es una "promesa"? ¿Qué es un plazo? Pero lo importante es ¿Todo esto está en la solicitud?

De veras que esto no es un trabajo sobre los préstamos. Si mencionas cosas como "penalización por incumplimiento" es porque eso forma parte del problema. Y forma parte del problema desde el minuto cero, porque no has excluido absolutamente nada de la historia de usuario.

Si lo que quieres es crear un fichero en un lenguaje de programación que contenga palabras como las de las historias de usuario que no signifiquen nada y sean inútiles para trabajar, es posible que consigas hacerlo así. Pero si lo que quieres es dividir el problema para solucionarlo, no parece que meter *toda* la información desde el minuto cero sea la mejor opción.

Reply:
Si la "exigencia" es simplemente "que todos los usuarios usen la misma fórmula", ni siquiera hay que mencionarlo. Basta simplemente con proporcionar una sola fórmula. Trata de no introducir información superflua, porque te recuerdo que estos son documentos de trabajo que tratan de responder a la pregunta "¿Qué hago ahora?"